src/webapp/models.py

```diff
@@ -479,11 +480,29 @@ def get_n_labels(n, difficulty_id):
         Reads all rows from database and chooses n random labels in a list.
     """
     try:
+
+        # get labels from three most recent games
+        games = Games.query.filter(
+            Games.difficulty_id <= difficulty_id,
+            difficulty_id
+            - Games.difficulty_id < 2).order_by(
+            Games.date.desc()).limit(3).all()
+        labels_to_filter = [
+            label for game in games for label in json.loads(game.labels)]
+
         # read all english labels in database
         labels = Labels.query.filter(
             Labels.difficulty_id <= difficulty_id).all()
         english_labels = [str(label.english) for label in labels]
+
+        minimum_labels = 6
```

Comment:
Semi-magic. If we want to have from last two games, maybe we want to set this to 
minimum_labels = 2*n
?

Reply:
OOps misunderstood